tesseract/tesseract_environment/src/ofkt/ofkt_state_solver.cpp

```diff
@@ -429,26 +433,52 @@ void OFKTStateSolver::onEnvironmentChanged(const Commands& commands)
         std::remove_if(joint_names_.begin(), joint_names_.end(), [removed_joints](const std::string& joint_name) {
           return (std::find(removed_joints.begin(), removed_joints.end(), joint_name) != removed_joints.end());
         }));
-    Eigen::MatrixX2d l1(limits_.rows() - static_cast<long>(removed_joints.size()), 2);
+
+    tesseract_common::KinematicLimits l1;
+    l1.joint_limits.resize(static_cast<long int>(joint_names_.size()), 2);
+    l1.velocity_limits.resize(static_cast<long int>(joint_names_.size()));
+    l1.acceleration_limits.resize(static_cast<long int>(joint_names_.size()));
+
     long cnt = 0;
-    for (long i = 0; i < limits_.rows(); ++i)
+    for (long i = 0; i < limits_.joint_limits.rows(); ++i)
+    {
       if (std::find(removed_joints_indices.begin(), removed_joints_indices.end(), i) == removed_joints_indices.end())
-        l1.row(cnt++) = limits_.row(i);
+      {
+        l1.joint_limits.row(cnt) = limits_.joint_limits.row(i);
+        l1.velocity_limits(cnt) = limits_.velocity_limits(i);
+        l1.acceleration_limits(cnt) = limits_.acceleration_limits(i);
+        ++cnt;
+      }
+    }
 
     limits_ = l1;
   }
 
   // Populate Joint Limits
-  if (new_limits.empty() == false)
+  if (new_kinematic_joints.empty() == false)
   {
-    Eigen::MatrixX2d l(limits_.rows() + static_cast<long>(new_limits.size()), 2);
-    Eigen::MatrixX2d l2(static_cast<long>(new_limits.size()), 2);
-    for (std::size_t i = 0; i < new_limits.size(); ++i)
+    tesseract_common::KinematicLimits l;
+    long s = limits_.joint_limits.rows() + static_cast<long>(new_kinematic_joints.size());
+    l.joint_limits.resize(s, 2);
+    l.velocity_limits.resize(s);
+    l.acceleration_limits.resize(s);
+
+    for (long i = 0; i < limits_.joint_limits.size(); ++i)
+    {
+      limits_.joint_limits.row(i) = limits_.joint_limits.row(i);
+      limits_.velocity_limits(i) = limits_.velocity_limits(i);
+      limits_.acceleration_limits(i) = limits_.acceleration_limits(i);
+    }
```

Comment:
These are both eigen types right? It might be quicker to just assign them rather than looping over and assigning each element

Reply:
Updated